hazelcast-sql/src/test/java/com/hazelcast/jet/sql_nightly/SqlSTSInnerEquiJoinFaultToleranceStressTest.java

```diff
@@ -116,10 +121,20 @@ public static void afterClass() {
 
     @Before
     public void setUp() throws Exception {
-        sqlService = instance().getSql();
+        // https://hazelcast.atlassian.net/browse/HZ-3187
+        if (Objects.equals(processingGuarantee, EXACTLY_ONCE) && !restartGraceful) {
+            logger.warning("Test skipped: https://hazelcast.atlassian.net/browse/HZ-3187");
+            return;
+        }
```

Comment:
Instead of having this condition in `@Before`, `@Test` and `@After`, you can just have the following in `@BeforeClass`:
```java
assumeFalse("Test skipped: https://hazelcast.atlassian.net/browse/HZ-3187",
        processingGuarantee.equals(EXACTLY_ONCE) && !restartGraceful);
```

Reply:
No, both `processingGuarantee` and `restartGraceful` are class members

Reply:
You can comment out `{EXACTLY_ONCE, false}` in `parameters()`. If what is skipped needs to be printed, you can print it in `@BeforeClass`.

Reply:
Burak, these checks will disappear after the fix of the mentioned issue. Does it have any difference how exactly did I disable the test?

Reply:
Functionally, it doesn't have any difference Sasha. But, are we reviewing only the functional side of changes? Isn't code elegancy or readability a concern?

Reply:
Very philosophical question. Code elegancy for a couple of LoC in tests, which will disappear in 1-2 months for me is not a concert. It is readable and reasonable. It includes a reason it was introduced.